src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,137 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+
+    private int size;
+
+    private Object[] dataArray;
+
+    public ArrayList() {
+        this.dataArray = new Object[START_CAPACITY];
+        this.size = 0;
+    }
+
+    private boolean sizeCheck() {
+        return size == dataArray.length;
+    }
+
+    private void grow(int minCapacity) {
```

Comment:
The `grow` method should be responsible for resizing the array only and should not accept a `minCapacity` parameter. Instead, the calculation for the new capacity should be done outside this method, and `grow` should only take the new capacity as an argument.